tensorflow/compiler/xla/service/gpu/gpu_compiler.cc

```diff
@@ -738,8 +738,6 @@ Status GpuCompiler::PrepareHloModuleForIrEmitting(HloModule* hlo_module) {
   pipeline.AddPass<CopyInsertion>(GetCanShareBuffer());
   // To fuse the copy.
   pipeline.AddPass<GpuHorizontalLoopFusion>("copy_");
-  // To remove temporary fused_computation created by GpuHorizontalLoopFusion
-  pipeline.AddPass<HloDCE>();
```

Comment:
What's the motivation behind removing that pass?

Reply:
It was discussed here: https://github.com/tensorflow/tensorflow/pull/57087#issuecomment-1276240281

Mostly, it was added as h-loop-fusion was not cleaning things up. But I and Adrian modified it to generates a clean graph. So it isn't needed anymore.

Reply:
Agree, this is not needed anymore :)